src/particles.cpp

```diff
@@ -445,8 +445,7 @@ void ParticleManager::step(float dtime)
 void ParticleManager::stepSpawners (float dtime)
 {
 	MutexAutoLock lock(m_spawner_list_lock);
-	for (std::map<u32, ParticleSpawner*>::iterator i =
-			m_particle_spawners.begin();
+	for (auto i = m_particle_spawners.begin();
```

Comment:
i think better candidate is: std::remove_if to replace this function with a secure method, no ? :)